### PR TITLE
chore: bump Go to version 1.20

### PR DESCRIPTION
**Pull request by odubajDT**
Fixes https://github.com/open-feature/go-sdk/issues/253

### CODE REVIEW

**Comment by Kavindu-Dodan** [comment]
My concern here is the version jump from 1.19 to 1.21. Given SDK is a central piece of OF, I think we should move step-wise (1.19 -> 1.20 -> 1.21) so that adaptation becomes easier where needed. We have a tracking issue for 1.20 already - https://github.com/open-feature/go-sdk/issues/253

**Review by Kavindu-Dodan — APPROVED** [review]

**Comment by Kavindu-Dodan** [comment]
I have upgraded the PR to bump Go version to 1.20. This allows us to upgrade SDK's go version linearly, avoiding unnecessary impact on downstream users.

**Review by Kavindu-Dodan — CHANGES_REQUESTED** [review]
Pending for TC decision on runtime support & documentation

**Comment by toddbaert** [comment]
After our meeting I'm sure I can say the TC will not block dropping Go 1.19, as long as the go-maintainers support that.

Go is a particularly fast-moving runtime.

**Review by toddbaert — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]

**Review by lukas-reining — APPROVED** [review]

**Comment by beeme1mr** [comment]
Can we extend the readme to include a section on our support policy?

**Comment by Kavindu-Dodan** [comment]
> Can we extend the readme to include a section on our support policy?

I tried to add this with this commit - https://github.com/open-feature/go-sdk/pull/255/commits/f854ec49213e6c980c35e0f77eafb31813db9f73

File: README.md [hunk 1]
```diff
@@ -43,7 +43,10 @@
 
 ### Requirements
 
-- Go 1.19+
+Go language version: [1.20](https://go.dev/doc/devel/release#go1.20)
+
+> [!NOTE]
+> OpenFeature Go SDK attempts to support currently maintained Go language version.
```
Comment (beeme1mr) [inline README.md:49]:
```suggestion
> The OpenFeature Go SDK only supports currently maintained Go language versions.
```

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**